### ISSUE TITLE
Error during reading file when generate code coverage report

### ISSUE THREAD

**Issue by chenlish** [danielpalme/ReportGenerator#610]
**Describe the bug**
Error during reading file: '[https://dev.azure.com/{repo}/_apis/git/repositories/{repo}/items?api-version=1.0&versionType=commit&version=xxxxxxx&path={file}']

**To Reproduce**
azure devops task
hosted latest linux 
net6.0
![image](https://github.com/danielpalme/ReportGenerator/assets/90018818/c319f05f-5906-4034-b87b-8988937532e8)

I have no idea on why reading file from a special commit when generate code coverage.
how can i fix this problem?

**Comment by danielpalme** [comment]
ReportGenerator reads the code files that appear in your `coverage.cobertura.xml`  file(s).
I guess one code file has the path 'https://dev.azure.com/{repo}/_apis/git/repositories/{repo}/items?api-version=1.0&versionType=commit&version=xxxxxxx&path={file}''.

Perhaps you can have a look at the `coverage.cobertura.xml`  file(s). Does is contain the url?

**Comment by chenlish** [comment]
> ReportGenerator reads the code files that appear in your `coverage.cobertura.xml` file(s). I guess one code file has the path '[https://dev.azure.com/{repo}/_apis/git/repositories/{repo}/items?api-version=1.0&versionType=commit&version=xxxxxxx&path={file}](https://dev.azure.com/%7Brepo%7D/_apis/git/repositories/%7Brepo%7D/items?api-version=1.0&versionType=commit&version=xxxxxxx&path=%7Bfile%7D)''.
> 
> Perhaps you can have a look at the `coverage.cobertura.xml` file(s). Does is contain the url?

thanks for your quick response. 
Yes, the 'coverage.cobertura.xml' file contains the url. like:

```
<class name="" filename="{url here}" line-rate="0" branch-rate="0" complexity="6">
```
because the git commit does not include those files, so it will throw errors here.
does this url generated by the reportGenerator?  how can i config the reportGenerator parameters to resolve it?

thanks.

**Comment by chenlish** [comment]
@danielpalme resolved. help me a lot. thanks.

**Comment by chenlish** [comment]
close it.

**Comment by AL71B** [comment]
I've had a similar issue, which I posted about under another issue ticket. https://github.com/danielpalme/ReportGenerator/issues/640

But this one seems more appropriate.

In summary, when our projects were targeting **.net 6**, we saw no issues from report generator. When we switched the projects to use **.net 8**, we saw the issues above "error reading file <URI> Response status code does not indicate success: 500 (Internal Server Error)" etc. 

Due to the issue reading the files, this in-turn caused problems with our SonarQube analysis steps. 

Thanks to the tips in this post, we looked at the Cobertura.xml files, and compared the .net 6 and .net 8 output. 

**.net 6 Cobertura.xml**
For each class listed, the xml showed `class name` and `filename`  but with filename being an actual file path i.e. no URI

**.net 8 Cobertura.xml**
Here, the xml showed `class name` and `filename` again, but the filename also contained the URI information. 
filename="<URI>;path=/path/to/file.cs"

This led us to question the `<UseSourceLink>true</UseSourceLink>` setting in our `cicd.runsettings` file. This has been set to `true` for a very long time and across a great number of projects - with no issues, but crucially, all projects at .net 6. 

We then set `<UseSourceLink>false</UseSourceLink>` and re-ran the build (.net 8) and there were no "Error reading file..." errors. 

So we have fixed the issue by changing the UseSourceLink setting. 

We tried different versions of the coverlet.collector, (version 3.1.2 and 6.0.2). We still needed to set UseSourceLink = false, for the Cobertura XML file to be generated in such a way that Report Generator could process it without the "Error reading file " message.

Hope this helps others that hit this problem.
